lib/plugins/aws/provider/awsProvider.js

```diff
@@ -208,8 +208,8 @@ class AwsProvider {
         if (_.has(event, 'http') && !validAPIGatewayStageNamePattern.test(stage)) {
           throw new Error([
             `Invalid stage name ${stage}: `,
-            'it should contains only [a-zA-Z0-9] for AWS provider if http event are present ',
-            'since API Gateway stage name cannot contains hyphens.',
+            'it should contains only [-a-zA-Z0-9] or Serverless variables for AWS provider',
```

Comment:
Shouldn't the message also include `{}$.`?

Reply:
Ok, Can you point me to right place

Reply:
@dschep why do you validate names locally? Why not let cloud formation run this check for you? The downside is fail-fast, but at least it won't be chasing AWS for changes. Does it make sense?